packages/lib/shared/hooks/usePeristedForm.ts

```diff
@@ -0,0 +1,38 @@
+import { useLocalStorage } from 'usehooks-ts'
```

Comment:
I have been thinking lately about adding more tests to the app (trying to refactor things around seems like a minefield sometimes). This looks like a good candidate to add some tests, but up to you.

Reply:
agree with the sentiment but want to explore building the transaction modal first to scope out the difficult parts
- #1128